gnucash/gnome/gnc-plugin-page-invoice.c

```diff
@@ -556,21 +651,21 @@ gnc_plugin_page_invoice_update_menus (GncPluginPage *page, gboolean is_posted, g
     gnc_plugin_update_actions (action_group, invoice_book_readwrite_actions,
                                "sensitive", !is_readonly);
 
-    for (i = 0; label_list[i].action_name; i++)
-    {
-        /* update the action labels */
-        action = gtk_action_group_get_action(action_group,
-                                             label_list[i].action_name);
-        gtk_action_set_label(action, _(label_list[i].label));
-    }
-
-    for (i = 0; tooltip_list[i].action_name; i++)
-    {
-        /* update the action tooltips */
-        action = gtk_action_group_get_action(action_group,
-                                             tooltip_list[i].action_name);
-        gtk_action_set_tooltip(action, _(tooltip_list[i].label));
-    }
+    /* update the action labels */
+    gnc_plugin_page_invoice_action_update (action_group, label_list, (void*)gtk_action_set_label);
+    /* update the action tooltips */
+    gnc_plugin_page_invoice_action_update (action_group, tooltip_list, (void*)gtk_action_set_tooltip);
+
+    // if there is no default layout do not enable reset action
+    layout_action = gnc_plugin_page_get_action (page, "WindowsResetLayoutAction");
+    if (gnc_invoice_window_has_default_state (priv->iw))
+        has_default = TRUE;
+    gtk_action_set_sensitive (layout_action, has_default);
```

Comment:
You're calling this code twice exactly like this. Perhaps spin it out into a helper function instead ?

Reply:
Will do.